hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -172,6 +172,9 @@ object HoodieSparkSqlWriter {
       operation = WriteOperationType.INSERT
     }
 
+    val sqlMergeIntoPrepped = parameters.getOrDefault(SPARK_SQL_MERGE_INTO_PREPPED_KEY, "false").toBoolean
+    val isPrepped = canDoPrepped(hoodieConfig, parameters, operation, df)
```

Comment:
let's just keep one boolean `isPrepped`? The boolean `sqlMergeIntoPrepped` can also be handled in `canDoPrepped`.

Reply:
HoodieCreateRecordUtils.createHoodieRecordRdd has some diff logic depending on whether its isPrepped or sqlMergeIntoPrepped. thats why